atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/CandidateSweepableBucketRetriever.java

```diff
@@ -18,14 +18,26 @@
 
 import com.palantir.atlasdb.sweep.asts.SweepStateCoordinator.SweepableBucket;
 import com.palantir.refreshable.Disposable;
-import java.util.List;
+import java.util.Set;
 import java.util.function.Consumer;
 
 // A push model - decouples asking for updates from the logic of debouncing requests.
 // Simplifies the coordinator, since it can just repeatedly _ask_ for updates whenever it thinks it's useful to do so
 // but we can rate limit it down.
 public interface CandidateSweepableBucketRetriever {
+    /**
+     * Requests an update to the set of buckets. This method will return immediately. There is no guarantee that
+     * a refresh will occur after the method is called, as this is simply a hint.
+     */
     void requestUpdate();
 
-    Disposable subscribeToChanges(Consumer<List<SweepableBucket>> task);
+    /**
+     * Registers a callback that will execute after each refresh, including whenever the set of buckets remains constant
+     * across refreshes.
+     * <p>
+     * Callbacks will be executed in the order they are registered. If a callback fails, subsequent callbacks will not
```

Comment:
seems fine, though is this the guarantee we want / will this be sufficient in the context of where this ends up being used?